packages/git/src/browser/git-widget.tsx

```diff
@@ -94,10 +98,10 @@ export class GitWidget extends GitDiffWidget implements StatefulWidget {
     }
 
     async initialize(repository: Repository | undefined): Promise<void> {
+        this.toDisposeOnInitialize.dispose();
```

Comment:
It will delete itself from all disposable collections, so you should to add it `this.toDispose` again on next line.

Reply:
please correct me if i am wrong:

having `this.toDispose.push(this.toDisposeOnInitialize)` in the `initialize()` function would push one reference of `this.toDisposeOnInitialize` to `this.toDispose` every time GitWidget refreshes, and that would potentially leave many references of the same thing (i.e., `this.toDisposeOnInitialize`) in `this.toDispose` - it does not hurt, but it is not necessary.

i did some debugging and found `this.toDisposeOnInitialize.dispose()` removed all elements from `this.toDisposeOnInitialize.disposables` array, and leave elements in `this.toDispose.disposables` untouched. so i guess this line will not delete itself from `this.toDispose`.

Reply:
`DisposableCollection` remove any disposable which gets disposed to avoid disposing it again: https://github.com/theia-ide/theia/blob/87ad22b5da0598fc9d40ee422b053457adbf8a4d/packages/core/src/common/disposable.ts#L86

Reply:
thank you for explaining ! i updated the PR.